Minifier/renamer.fs

```diff
@@ -151,29 +155,30 @@ type private RenamerVisitor(options: Options.Options, level: Level) =
             renExpr env cond
             env
         | ForD(init, cond, inc, body) as stmt ->
-            let newEnv = env.onEnterScope env stmt
-            let newEnv = renDecl false newEnv init
-            renStmt newEnv body |> ignore<Env>
-            Option.iter (renExpr newEnv) cond
-            Option.iter (renExpr newEnv) inc
-            if options.hlsl then newEnv
+            let innerEnv = env.onEnterScope env stmt // In the for scope, we use an env that allows shadowing unused outer decls.
+            let envForType = Some env // Use the inner env to rename variables, but use the outer env to rename the variable type!
```

Comment:
Why do we need the outer env for the type?

Reply:
Same reason as for function args. Inside the block, the type might be unused and shadowed.